roles/identity-management/manage-rh-sso-identities/tasks/create_users.yml

```diff
@@ -0,0 +1,48 @@
+---
+
+- name: "Get Red Hat SSO access token"
+  include_tasks: access-token.yml
+
+- name: "Retrieve template for user manipulation"
+  set_fact:
+    sso_user_json: "{{ lookup('template', 'manage-users.json.j2') | to_json }}"
+
+- name: "Get existing users"
```

Comment:
Seems quite "expensive" to do this on every user. One improvement here would be to do this step as a `pre` task and filter out which users need to be added/deleted, and hence only have to do this once.

Reply:
That makes sense. I fixed this problem by using REST API query parameters, that way I won't need to get a list of users and search for the desired ID from that. Instead, I am now retrieving the desired user parameters to capture the ID.
Additionally, I am separating the pre-requisites into a separate task as you suggested. This will help me to re-use some of these resources when manipulating groups.
Thanks for the comments.

Reply:
OK, that's a good change, but potentially still a bit too expensive than it have to be. As mentioned before, please open an issue so we can address in a follow-up .